include/EASTL/tuple.h

```diff
@@ -145,6 +145,41 @@ class tuple_element<I, const volatile Internal::TupleImpl<Indices, Ts...>> : pub
 {
 };
 
+// attempt to isolate index given a type
+template <typename T, typename Tuple>
+struct tuple_index
+{
+};
+
+template <typename T>
+struct tuple_index<T, TupleTypes<>>
+{
+	typedef void DuplicateTypeCheck;
+	tuple_index() = delete; // tuple_index should only be used for compile-time assistance, and never be instantiated
+	static const size_t index = 0;
+};
+
+template <typename T, typename... Ts>
+struct tuple_index<T, TupleTypes<T, Ts...>>
+{
+	typedef int DuplicateTypeCheck;
+	static_assert(is_void<typename tuple_index<T, TupleTypes<Ts...>>::DuplicateTypeCheck>::value, "duplicate type T in tuple_vector::get<T>(); unique types must be provided in declaration, or only use get<size_t>()");
```

Comment:
I'd leave a comment here explaining how we are marking the recursive template instantiations to ensure there are not duplicates.